clang/lib/Index/IndexRecordHasher.cpp

```diff
@@ -43,6 +43,8 @@ class DeclHashVisitor : public ConstDeclVisitor<DeclHashVisitor, hash_code> {
     if (auto *attr = D->getExternalSourceSymbolAttr()) {
       COMBINE_HASH(hash_value(attr->getDefinedIn()));
     }
+    if (D->getDeclName().isEmpty())
+      return Hash;
     return COMBINE_HASH(Hasher.hash(D->getDeclName()));
```

Comment:
For better future maintenance I would recommend to instead change `hash_code IndexRecordHasher::hash(DeclarationName Name)` method to return `INITIAL_HASH` if the name is empty.

Reply:
Thanks, updated